src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx

```diff
@@ -1,47 +1,81 @@
-import { CustomEmoji, Data, EmojiData, NimblePicker } from 'emoji-mart'
-import emojiData from 'emoji-mart/data/twitter.json'
-import React, { useRef } from 'react'
+import { Picker } from 'emoji-picker-element'
+import { CustomEmoji, EmojiClickEvent, EmojiClickEventDetail } from 'emoji-picker-element/shared'
+import React, { useCallback, useEffect, useMemo, useRef } from 'react'
+import { useSelector } from 'react-redux'
 import { useClickAway } from 'react-use'
-import { ShowIf } from '../../../../common/show-if/show-if'
+import { ApplicationState } from '../../../../../redux'
 import './emoji-picker.scss'
 import forkawesomeIcon from './forkawesome.png'
 import { ForkAwesomeIcons } from './icon-names'
 
 export interface EmojiPickerProps {
   show: boolean
-  onEmojiSelected: (emoji: EmojiData) => void
+  onEmojiSelected: (emoji: EmojiClickEventDetail) => void
   onDismiss: () => void
 }
 
 export const customEmojis: CustomEmoji[] = Object.keys(ForkAwesomeIcons).map((name) => ({
   name: `fa-${name}`,
-  short_names: [`fa-${name.toLowerCase()}`],
-  text: '',
-  emoticons: [],
-  keywords: ['fork awesome'],
-  imageUrl: forkawesomeIcon,
-  customCategory: 'ForkAwesome'
+  shortcodes: [`fa-${name.toLowerCase()}`],
+  url: forkawesomeIcon,
+  category: 'ForkAwesome'
 }))
 
 export const EmojiPicker: React.FC<EmojiPickerProps> = ({ show, onEmojiSelected, onDismiss }) => {
-  const pickerRef = useRef(null)
+  const darkModeEnabled = useSelector((state: ApplicationState) => state.darkMode.darkMode)
+  const pickerContainerRef = useRef<HTMLDivElement>(null)
+  const firstOpened = useRef(false)
 
-  useClickAway(pickerRef, () => {
+  useClickAway(pickerContainerRef, () => {
     onDismiss()
   })
 
+  const emojiClickListener = useCallback((event) => {
+    onEmojiSelected((event as EmojiClickEvent).detail)
+  }, [onEmojiSelected])
+
+  const twemojiStyle = useMemo(() => {
+    const style = document.createElement('style')
+    style.textContent = 'section.picker { --font-family: "twemoji" !important; }'
+    return style
+  }, [])
+
+  useEffect(() => {
+    if (!pickerContainerRef.current || firstOpened.current) {
+      return
+    }
+    const picker = new Picker({
+      customEmoji: customEmojis,
+      dataSource: '/static/js/emoji-data.json'
+    })
+    const container = pickerContainerRef.current
+    picker.addEventListener('emoji-click', emojiClickListener)
```

Comment:
remove this listener onUnload?

Reply:
This doesn't work unfortunately. Emojis get unselectable from the picker when applying the following change:

```diff
index 3b98108..d42c743 100644
--- a/src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx
+++ b/src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx
@@ -55,6 +55,9 @@ export const EmojiPicker: React.FC<EmojiPickerProps> = ({ show, onEmojiSelected,
     }
     container.appendChild(picker)
     firstOpened.current = true
+    return () => {
+      picker.removeEventListener('emoji-click', emojiClickListener)
+    }
   }, [pickerContainerRef, emojiClickListener, darkModeEnabled, twemojiStyle])

   useEffect(() => {
```

Reply:
Apparently the useEffect expires already when the picker is opened.

Reply:
okay then I'll close this